apis/inv/v1alpha1/schema_helpers.go

```diff
@@ -66,9 +68,20 @@ func (r *SchemaSpec) GetNewSchemaBase(basePath string) SchemaSpecSchema {
 	basePath = r.GetBasePath(basePath)
 
 	// when no models are supplied we use the base dir
-	models := initSlice(r.Schema.Models, ".")
-	includes := initSlice(r.Schema.Includes, "")
-	excludes := initSlice(r.Schema.Excludes, "")
+	modelsSet := sets.New[string]()
+	includesSet := sets.New[string]()
+	excludesSet := sets.New[string]()
+	for _, repo := range r.Repositories {
+		mergeSetWithSlice(modelsSet, repo.Schema.Models)
+		mergeSetWithSlice(includesSet, repo.Schema.Includes)
+		mergeSetWithSlice(excludesSet, repo.Schema.Excludes)
```

Comment:
I would replace this with the following for readability, and according to the documentation Set is implemented as a map so there is no need to check if the set contains the item first:
```suggestion
		modelsSet.Insert(repo.Schema.Models...)
		includesSet.Insert(repo.Schema.Includes...)
		excludesSet.Insert(repo.Schema.Excludes...)
```